Extension/src/Utility/Text/scanner.ts

```diff
@@ -0,0 +1,1018 @@
+                    // FYI:
+                    // Well-known characters that are currently not processed
+                    //   # \
+                    // will need to update the scanner if there is a need to recognize them
+                    return isIdentifierStart(this.#ch) ? this.scanIdentifier() : this.next(Kind.Unknown);
+            }
+        }
+
+        this.text = '';
+        return this.kind = Kind.EndOfFile;
+    }
+
+    take(): Token {
+        const result = { ...this };
+        this.scan();
+        return result;
+    }
+
+    *takeUntil(endToken: Kind, options?: { escape?: Kind[]; nestable?: [Kind, Kind][] }, yieldFinalClose?: boolean): Iterable<Token> {
+        const nestable = options?.nestable || [];
+        const escape = options?.escape || [];
+
+        processing: do {
+            switch (this.kind) {
+                case endToken:
+                    // if we're nested, we need to return the end token, because it's significant to the consumer
+                    if (yieldFinalClose) {
+                        yield this.take();
+                        return;
+                    }
+
+                    // we're done here, lose the last token and get out
+                    this.take();
+                    return;
+
+                case Kind.EndOfFile:
+                    throw new Error('Unexpected end of tokens');
+            }
+
+            // check for escaped tokens
+            if (escape.includes(this.kind)) {
+                // pull thru the escape token
+                // and the next token
+                yield this.take();
+                yield this.take();
+                continue;
+            }
+
+            // check for nested tokens
+            for (const [open, close] of nestable as [Kind, Kind][]) {
+                if (this.kind === open) {
+                    yield this.take(); // yield the open token
+                    yield* this.takeUntil(close, options, true);
+                    continue processing;
+                }
+            }
+
+            // yield the current token
+            yield this.take();
+        } while (true);
+    }
+
+    takeWhitespace() {
+        while (!this.eof && this.isWhitespace) {
+            this.scan();
+        }
+    }
+
+    takeWhiteSpaceAndNewLines() {
+        while (!this.eof && (this.isWhitespace || this.isNewLine)) {
+            this.scan();
+        }
+    }
+
+    get isWhitespace() {
+        return this.kind === Kind.Whitespace;
+    }
+
+    get isNewLine() {
+        return this.kind === Kind.NewLine;
+    }
+
+    get isEndOfFile() {
+        return this.kind === Kind.EndOfFile;
+    }
+
+    get isComment() {
+        switch (this.kind) {
+            case Kind.SingleLineComment:
+            case Kind.MultiLineComment:
+            case Kind.SingleLineHashComment:
+            case Kind.MultiLineHashComment:
+                return true;
+        }
+        return false;
+    }
+
+    /**

```

Comment:
The `/*` appears to have an extra space before it.

Reply:
fixing